Tests/Tests/APIClientSpec.swift

```diff
@@ -121,8 +143,10 @@ class APIClientSpec: QuickSpec {
           let client = APIClient(requestPerformer: performer)
           var error: SwishError?
 
-          client.perform(request) {
-            error = $0.error
+          client.perform(request) { result in
+            if case let .failure(resultError) = result {
```

Comment:
I think this pattern could also be tidied up with `result.get()`:

```swift
var result: Result<Foo, SwishError>?

client.perform(request) {
  result = $0
}

expect { try result?.get() }.toEventually(throwError(SwishError.serverError(code: expectedCode, data: performer.data)))
```

Reply:
This one does not appear to work. The function given to expect is expected to produce a `String?`, so the `Predicate` fails to line up type-wise.

Reply:
Ah sorry I was using `Foo` as an example because I didn’t know what the actual type would be. This compiles for me:

```swift
          var result: Result<String, SwishError>?

          client.perform(request) {
            result = $0
          }

          expect { try result?.get() }.toEventually(throwError(SwishError.serverError(code: expectedCode, data: performer.data)))
```

Reply:
Ohh I see, the detail is you can expect to throw an error, not just expect to get an error our.